File: backend/payment/src/main/java/com/bbmovie/payment/entity/UserSubscription.java
package com.bbmovie.payment.entity;

import com.bbmovie.payment.entity.base.BaseEntity;
import com.bbmovie.payment.entity.enums.PaymentProvider;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@Entity
@Table(name = "user_subscriptions")
@NoArgsConstructor
@AllArgsConstructor
public class UserSubscription extends BaseEntity {

    @Column(name = "user_id", nullable = false)
    private String userId;

    @Column(name = "user_email", nullable = false)
    private String userEmail;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "plan_id", nullable = false)
    private SubscriptionPlan plan;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Builder.Default
    @Column(name = "is_active")
    private boolean isActive = false;

    @Builder.Default
    @Column(name = "auto_renew")
    private boolean autoRenew = true;

    @Column(name = "last_payment_date")
    private LocalDateTime lastPaymentDate;

    @Column(name = "next_payment_date")
    private LocalDateTime nextPaymentDate;

    @Column(name = "payment_provider")
    @Enumerated(EnumType.STRING)
    private PaymentProvider paymentProvider;

    @Column(name = "payment_method")
    private String paymentMethod;
}

File: backend/payment/src/main/kotlin/com/bbmovie/payment/entity/UserSubscription.kt
package com.bbmovie.payment.entity

import com.bbmovie.payment.entity.base.BaseEntity
import jakarta.persistence.*
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "user_subscriptions")
open class UserSubscription : BaseEntity() {

    @Column(name = "user_id", nullable = false)
    open lateinit var user: UUID

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "plan_id", nullable = false)
    open lateinit var plan: SubscriptionPlan

    @Column(name = "start_date", nullable = false)
    open lateinit var startDate: LocalDateTime

    @Column(name = "end_date", nullable = false)
    open lateinit var endDate: LocalDateTime

    @Column(name = "is_active")
    open var isActive: Boolean = true

    @Column(name = "auto_renew")
    open var autoRenew: Boolean = false

    @Column(name = "last_payment_date")
    open var lastPaymentDate: LocalDateTime? = null

    @Column(name = "next_payment_date")
    open var nextPaymentDate: LocalDateTime? = null

    @Column(name = "payment_provider")
    @Enumerated(EnumType.STRING)
    open var paymentProvider: PaymentProvider? = null

    @Column(name = "payment_gateway_id")
    open var paymentGatewayId: String? = null
}